File: csharp/Agentic.Commerce.Protocol/Models/RefundType.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Agentic.Commerce.Protocol
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum RefundType
    {
        [EnumMember(Value = "store_credit")]
        StoreCredit = 1,

        [EnumMember(Value = "original_payment")]
        OriginalPayment = 2,
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/RefundType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class RefundType : ToQueryParam {
    @SerialName("store_credit") STORE_CREDIT,
    @SerialName("original_payment") ORIGINAL_PAYMENT;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
